CRM/Contribute/BAO/Contribution.php

```diff
@@ -4715,14 +4715,10 @@ public static function sendMail(&$input, &$ids, $contributionID, &$values, $recu
     $contribution->loadRelatedObjects($input, $ids, TRUE);
     // set receipt from e-mail and name in value
     if (!$returnMessageText) {
-      $userID = CRM_Core_Session::singleton()->getLoggedInContactID();
-      if (!empty($userID)) {
-        list($userName, $userEmail) = CRM_Contact_BAO_Contact_Location::getEmailDetails($userID);
-        $values['receipt_from_email'] = CRM_Utils_Array::value('receipt_from_email', $input, $userEmail);
-        $values['receipt_from_name'] = CRM_Utils_Array::value('receipt_from_name', $input, $userName);
-      }
+      list($fromName, $fromEmail) = self::generateFromEmailAndName($input, $contribution);
```

Comment:
This seems a little clumsy - why not either list directly to the $values['receipt_from_name'] or array_merge the results into $values